inc/targetbase.class.php

```diff
@@ -764,4 +773,76 @@ protected function parseTags($content, PluginFormcreatorForm_Answer $formanswer,
 
       return $content;
    }
+
+
+   protected function showLocationSettings($rand) {
+      global $DB;
+
+      echo '<tr class="line0">';
+      echo '<td width="15%">' . __('Location') . '</td>';
+      echo '<td width="45%">';
+      Dropdown::showFromArray('location_rule', static::getEnumLocationRule(), array(
+         'value'                 => $this->fields['location_rule'],
+         'on_change'             => 'change_location()',
+         'rand'                  => $rand
+      ));
+      $script = <<<EOS
+         function change_location() {
+            $('#location_specific_title').hide();
+            $('#location_specific_value').hide();
+            $('#location_question_title').hide();
+            $('#location_question_value').hide();
+
+            switch($('#dropdown_location_rule$rand').val()) {
+               case 'answer' :
+                  $('#location_question_title').show();
+                  $('#location_question_value').show();
+                  break;
+               case 'specific':
+                  $('#location_specific_title').show();
+                  $('#location_specific_value').show();
+                  break;
+            }
+         }
+         change_location();
+EOS;
+      echo Html::scriptBlock($script);
+      echo '</td>';
+      echo '<td width="15%">';
+      echo '<span id="location_question_title" style="display: none">' . __('Question', 'formcreator') . '</span>';
+      echo '<span id="location_specific_title" style="display: none">' . __('Location ', 'formcreator') . '</span>';
+      echo '</td>';
+      echo '<td width="25%">';
+
+      echo '<div id="location_specific_value" style="display: none">';
+      Location::dropdown(array(
+         'name' => '_location_specific',
+         'value' => $this->fields["location_question"],
+      ));
+      echo '</div>';
+      echo '<div id="location_question_value" style="display: none">';
+      // select all user questions (GLPI Object)
+      $query2 = "SELECT q.id, q.name, q.values
```

Comment:
``DBmysqlIterator`` possible ?

Reply:
I planned to cleanup the code to 
- remove warnings raised by my IDE, 
- replace array() by [], 
- drop remaining raw SQL

one PR per file or small group of files

This code was done before I was sure to drop the compatibility with GLPI 9.1